stream_alert_cli/package.py

```diff
@@ -160,66 +163,67 @@ def _sha256sum(package_path):
+        third_party_libs = self.config[self.config_key]['third_party_libraries']
         if third_party_libs:
             if len(third_party_libs) > 0:
-                logging.info('Installing third-party libraries')
+                LOGGER_CLI.info('Installing third-party libraries')
                 pip_command = ['install']
                 pip_command.extend(third_party_libs)
                 pip_command.extend(['--upgrade', '--target', temp_package_path])
                 pip.main(pip_command)
             else:
-                logging.info('No third-party libraries to install.')
+                LOGGER_CLI.info('No third-party libraries to install.')
 
     def _upload(self, package_path):
-        """Upload the StreamAlert package and sha256 to S3."""
-        logging.info('Uploading StreamAlert package to S3')
-        client = boto3.client('s3', region_name=self.config['region'])
+        """Upload the StreamAlert package and sha256 to S3.
+
+        Args:
+            package path [string]:  Full path to the zipped dpeloyment package
+
+        Returns:
+            [boolean] Indicating a successful upload
+
+        Raises:
+            Exception if client put_object fails
+        """
+        LOGGER_CLI.info('Uploading StreamAlert package to S3')
+        client = boto3.client('s3', region_name=self.config['account']['region'])
         # the zip and the checksum file
         for package_file in (package_path, '{}.sha256'.format(package_path)):
             package_name = package_file.split('/')[-1]
             package_fh = open(package_file, 'r')
             try:
                 client.put_object(
-                    Bucket=self.config['lambda_source_bucket_name'],
-                    Key=os.path.join(self.source_prefix, package_name),
+                    Bucket=self.config[self.config_key]['source_bucket'],
+                    Key=os.path.join(self.package_name, package_name),
                     Body=package_fh,
                     ServerSideEncryption='AES256'
                 )
-            except:
-                logging.info('An error occured while uploding %s', package_name)
+            except BaseException:
+                LOGGER_CLI.info('An error occured while uploding %s', package_name)
                 raise
             package_fh.close()
-            logging.info('Uploaded %s to S3', package_name)
+            LOGGER_CLI.info('Uploaded %s to S3', package_name)
 
         return True
 
-class AlertPackage(LambdaPackage):
-    """Deployment package class for the AWS Lambda `Alert` function"""
-    package_folders = {'stream_alert', 'rules', 'conf'}
-    package_files = {'main.py', 'variables.json'}
+
+class RuleProcessorPackage(LambdaPackage):
+    """Deployment package class for the StreamAlert Rule Processor function"""
+    package_folders = {'stream_alert/rule_processor', 'rules', 'conf'}
+    package_files = {'stream_alert/__init__.py', 'variables.json'}
+    package_root_dir = '.'
```

Comment:
They both weren't always `'.'`, and they may change in the future, so I'm going to leave it